docs/docs/core/00-baseapp.md

```diff
@@ -223,7 +223,40 @@ The consensus engine handles two main tasks:
 
 It is **not** the role of the consensus engine to define the state or the validity of transactions. Generally, transactions are handled by the consensus engine in the form of `[]bytes`, and relayed to the application via the ABCI to be decoded and processed. At keys moments in the networking and consensus processes (e.g. beginning of a block, commit of a block, reception of an unconfirmed transaction, ...), the consensus engine emits ABCI messages for the state-machine to act on.
 
-Developers building on top of the Cosmos SDK need not implement the ABCI themselves, as `BaseApp` comes with a built-in implementation of the interface. Let us go through the main ABCI messages that `BaseApp` implements: [`CheckTx`](#checktx) and [`DeliverTx`](#delivertx)
+Developers building on top of the Cosmos SDK need not implement the ABCI themselves, as `BaseApp` comes with a built-in implementation of the interface. Let us go through the main ABCI messages that `BaseApp` implements:
+
+* [`Prepare Proposal`](#prepare-proposal)
+* [`Process Proposal`](#process-proposal)
+* [`CheckTx`](#checktx)
+* [`DeliverTx`](#delivertx)
+
+
+### Prepare Proposal
+
+### Process Proposal
```

Comment:
```suggestion
### Process Proposal

```

Reply:
Headers should have new lines after them :P

Reply:
Let's do that in part 2 👍